### ISSUE TITLE
Firebase Crashlytics not submitting reports automatically

### ISSUE THREAD

**Issue by ski081** [firebase/firebase-ios-sdk#5805]
### Step 0: Are you in the right place?

  * For issues or feature requests related to __the code in this repository__
    file a Github issue.
    * If this is a __feature request__ please use the Feature Request template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the `firebase` tag.
  * For general (non-iOS) Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For backend issues, console issues, and other non-SDK help that does not fall under one
    of the above categories, reach out to
    [Firebase Support](https://firebase.google.com/support/).
  * Once you've read this section and determined that your issue is appropriate for
    this repository, please delete this section.

### [REQUIRED] Step 1: Describe your environment

  * Xcode version: 11.3.1
  * Firebase SDK version: 6.15.0
  * Firebase Component: FirebaseCrashlytics
  * Component version: 6.15.0
  * Installation method: `CocoaPods`

### [REQUIRED] Step 2: Describe the problem
I just upgraded from the Fabric version of Crashlytics to FirebaseCrashlytics. I followed the upgrade guide, downloaded the new **GoogleService-Info.plist**, updated my Run Script Build Phase, and updated my Swift code to initialize Firebase:

```
FirebaseApp.configure()
```

I switched to crashing my app as a test using the following:

```
fatalError("testing crash")
```

I also switched my debug configuration to use the **DWARF with dSYM File** in the **Debug Information Format** setting. When I crash the app, the report is not created in the Crashlytics console on Firebase. 

When I added the following code, however, crash reports started showing up:

```
Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(false)

Crashlytics.crashlytics().checkForUnsentReports { _ in
    Crashlytics.crashlytics().sendUnsentReports()
}
```

I have tried this both on simulator and from a device attached and not attached to the debugger.

#### Steps to reproduce:

1. Force crash the app using `fatalError()`
2. Relaunch the app
3. Check Firebase console for the associated crash

**Comment by iRaru** [comment]
@ski081 Try to launch your app without Xcode directly from the device. And after crash relaunch the app and check your dashboard.

For me, it works like this. If I build the app from Xcode and crash, then I cannot see those crashes.

**Comment by ski081** [comment]
Hi @iRaru. I tried running from Xcode directly to device, and even tried running without Xcode attached from the device. I'm getting the same results both ways. The crash happens, but unless I submit its via code (from he snippet above), I am not seeing crashes appear on the console.

**Comment by iRaru** [comment]
@ski081 did you uploaded dSYM files too? Check the missing dSYM tab for your version.

**Comment by RedDragonJ** [comment]
I tried this as well and it also behaves the same. But I wonder if this only happen in the development environment or not?!

**Comment by samedson** [comment]
I'm wondering if there's another setting that's impacting you. If you check out the documentation for [checkForUnsentReports](https://firebase.google.com/docs/reference/swift/firebasecrashlytics/api/reference/Classes/Crashlytics#checkforunsentreportscompletion:) we explain the criteria for turning off automatic collection. I'm wondering if maybe you have `FirebaseCrashlyticsCollectionEnabled` in your Info.plist, or if at some point you set FIRApp's `isDataCollectionDefaultEnabled` to false.

Otherwise, you can probably work around this by setting `Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(true)`. It will persist across runs of the app.

**Comment by ski081** [comment]
@samedson Verified the following things:

- We do not have `FirebaseCrashlyticsCollectionEnabled` in our Info.plist
- We are not setting `isDataCollectionDefaultEnabled` to false anywhere

I tried calling `Crashlytics.crashlytics().setCrashlyticsCollectionEnabled(true)` on app startup. This seems to be working, so I'm able to remove the manual upload code I referenced in this issue. Thank you for your help!
